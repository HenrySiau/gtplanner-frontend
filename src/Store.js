import React, { Component } from 'react';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import reducers from './reducers';
import { ConnectedRouter, routerMiddleware } from 'react-router-redux';
import createHistory from 'history/createBrowserHistory';


const loggerMiddleware = createLogger();
// Create a history of your choosing (we're using a browser history in this case)
const history = createHistory();
// Build the middleware for intercepting and dispatching navigation actions
const boundRouterMiddleware = routerMiddleware(history);

let store = createStore(
    reducers,
    applyMiddleware(
        thunkMiddleware,
        loggerMiddleware,
        boundRouterMiddleware
    )
);
export default class Store extends Component {
    render() {
        return (
            <Provider store={store}>
                { /* ConnectedRouter will use the store from Provider automatically */}
                <ConnectedRouter history={history}>
                {this.props.children}
                </ConnectedRouter>
            </Provider>
        )
    }
}


