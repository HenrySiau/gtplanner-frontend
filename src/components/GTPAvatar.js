import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import IconButton from 'material-ui/IconButton';
import Avatar from 'material-ui/Avatar';
import settings from '../config';
import { MenuItem, MenuList } from 'material-ui/Menu';
import { connect } from 'react-redux';
import { logout } from '../actions';
import { push } from 'react-router-redux';
import Popover from 'material-ui/Popover';

const styles = theme => ({
    avatar: {
        // margin: 10,
        width: 36,
        height: 36,
        marginRight: theme.spacing.unit * 2,
    },
});


class GTPAvatar extends React.Component {

    state = {
        avatarAnchor: null,
    };


    handleClick = event => {
        this.setState({ avatarAnchor: event.currentTarget });
    };

    handleClose = () => {
        this.setState({ avatarAnchor: null });
    };


    render() {
        const { classes } = this.props;
        const { avatarAnchor } = this.state;

        return (
            <div >

                <IconButton
                    id='appBarAvatar'
                    onClick={this.handleClick}>
                    <Avatar
                        alt="Profile Photo"
                        src={settings.imageServerUrl + '/images/user.png'}
                        className={classes.avatar}
                    /> </IconButton >

                <Popover
                    anchorEl={avatarAnchor}
                    open={Boolean(avatarAnchor)}
                    onClose={this.handleClose}
                    anchorOrigin={{ horizontal: 'left', vertical: 'bottom' }}
                    transformOrigin={{ horizontal: 'left', vertical: 'top' }}
                >
                    <MenuList role="menu">
                        <MenuItem onClick={() => {
                            this.props.dispatch(push('/myaccount'));
                            this.setState({ avatarAnchor: null });
                        }}>My account</MenuItem>
                        <MenuItem onClick={() => {
                            this.props.dispatch(push('/help'));
                            this.setState({ avatarAnchor: null });
                        }}>Help</MenuItem>
                        <MenuItem onClick={() => {
                            this.props.dispatch(logout);
                            this.setState({ avatarAnchor: null });
                            window.location = "/";
                        }}>Logout</MenuItem>
                    </MenuList>
                </Popover>
            </div>
        )
    }
}

GTPAvatar.propTypes = {
    classes: PropTypes.object.isRequired,
};
GTPAvatar = withStyles(styles)(GTPAvatar);
export default connect()(GTPAvatar)