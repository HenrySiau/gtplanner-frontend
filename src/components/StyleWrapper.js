import React, { Component } from 'react';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import DateFnsUtils from 'material-ui-pickers/utils/date-fns-utils';
import MuiPickersUtilsProvider from 'material-ui-pickers/utils/MuiPickersUtilsProvider';
import CssBaseline from '@material-ui/core/CssBaseline';

const theme = createMuiTheme({
    palette: {
        primary: {
            light: '#6ec6ff',
            main: '#2196f3',
            dark: '#0069c0',
            contrastText: '#fff',
        },
        secondary: {
            light: '#ff7961',
            main: '#f44336',
            dark: '#ba000d',
            contrastText: '#000',
        },
    },
});


export default class StyleWrapper extends Component {
    render() {
        const { classes } = this.props;
        return (
            <MuiThemeProvider theme={theme}>
                <MuiPickersUtilsProvider utils={DateFnsUtils}>
                    {/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
                    <CssBaseline />
                    {this.props.children}
                </MuiPickersUtilsProvider>
            </MuiThemeProvider>
        );
    }
}

