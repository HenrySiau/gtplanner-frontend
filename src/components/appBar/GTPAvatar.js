import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import Avatar from '@material-ui/core/Avatar';
import MenuItem from '@material-ui/core/MenuItem';
import MenuList from '@material-ui/core/MenuList';
import Popover from '@material-ui/core/Popover';

const styles = theme => ({
    avatar: {
        width: 36,
        height: 36,
    },
    avatarButton: {
        marginRight: theme.spacing.unit * 2,
    }
});

class GTPAvatar extends React.Component {

    state = {
        avatarAnchor: null,
    };

    handleClick = event => {
        this.setState({ avatarAnchor: event.currentTarget });
    };

    handleClose = () => {
        this.setState({ avatarAnchor: null });
    };

    render() {
        const { classes } = this.props;
        const { avatarAnchor } = this.state;

        return (
            <div >

                <IconButton
                    id='appBarAvatar'
                    onClick={this.handleClick}
                    className={classes.avatarButton}
                    >
                    <Avatar
                        alt="Profile Photo"
                        src={this.props.profilePictureURL}
                        className={classes.avatar}
                    /> </IconButton >

                <Popover
                    anchorEl={avatarAnchor}
                    open={Boolean(avatarAnchor)}
                    onClose={this.handleClose}
                    anchorOrigin={{ horizontal: 'left', vertical: 'bottom' }}
                    transformOrigin={{ horizontal: 'left', vertical: 'top' }}
                >
                    <MenuList role="menu">
                        <MenuItem onClick={() => {
                            this.props.push('/myaccount');
                            this.setState({ avatarAnchor: null });
                        }}>My account</MenuItem>
                        <MenuItem onClick={() => {
                            this.props.snackbarMessage('Helping Center Under Construction');
                            this.setState({ avatarAnchor: null });
                        }}>Help</MenuItem>
                        <MenuItem onClick={() => {
                            this.props.logout();
                            this.setState({ avatarAnchor: null });
                            window.location = "/";
                        }}>Logout</MenuItem>
                    </MenuList>
                </Popover>
            </div>
        )
    }
}
GTPAvatar.defaultProps = {
    profilePictureURL: '/defaultUserIcon.png',
};

GTPAvatar.propTypes = {
    classes: PropTypes.object.isRequired,
    profilePictureURL: PropTypes.string.isRequired,
    push: PropTypes.func,
    logout: PropTypes.func,
};

export default GTPAvatar = withStyles(styles)(GTPAvatar);