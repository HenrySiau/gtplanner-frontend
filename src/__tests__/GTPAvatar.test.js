import React from 'react';
// import GTPAvatar from '../components/appBar/GTPAvatar';
import GTPAvatar from '../components/appBar/GTPAvatar';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Store from '../Store';
import StyleWrapper from '../components/StyleWrapper';

configure({ adapter: new Adapter() })

test('Avatar without profilePictureURL', () => {
    const component = renderer.create(
        <Store>
            <StyleWrapper>
                <GTPAvatar
                />
            </StyleWrapper>
        </Store>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Avatar with profilePictureURL', () => {
    const component = renderer.create(
        <Store>
            <StyleWrapper>
                <GTPAvatar
                profilePictureURL='/henry-XXX-UserIcon.png'
                />
            </StyleWrapper>
        </Store>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});