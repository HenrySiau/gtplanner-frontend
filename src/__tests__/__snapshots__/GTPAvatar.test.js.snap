// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar with profilePictureURL 1`] = `
<div>
  <button
    className="MuiButtonBase-root-9 MuiIconButton-root-3 GTPAvatar-avatarButton-2"
    disabled={false}
    id="appBarAvatar"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiIconButton-label-8"
    >
      <div
        className="MuiAvatar-root-12 GTPAvatar-avatar-1"
      >
        <img
          alt="Profile Photo"
          className="MuiAvatar-img-14"
          sizes={undefined}
          src="/henry-XXX-UserIcon.png"
          srcSet={undefined}
        />
      </div>
       
    </span>
    <span
      className="MuiTouchRipple-root-15"
    />
  </button>
</div>
`;

exports[`Avatar without profilePictureURL 1`] = `
<div>
  <button
    className="MuiButtonBase-root-9 MuiIconButton-root-3 GTPAvatar-avatarButton-2"
    disabled={false}
    id="appBarAvatar"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiIconButton-label-8"
    >
      <div
        className="MuiAvatar-root-12 GTPAvatar-avatar-1"
      >
        <img
          alt="Profile Photo"
          className="MuiAvatar-img-14"
          sizes={undefined}
          src="/defaultUserIcon.png"
          srcSet={undefined}
        />
      </div>
       
    </span>
    <span
      className="MuiTouchRipple-root-15"
    />
  </button>
</div>
`;
